analyses/mdg/urban_exploration.md

```diff
@@ -0,0 +1,59 @@
+### Urban classification in Madagascar
+
+Very simple script to classify ADM4 areas in Madagascar as urban based on GHS data.
+
+```python
+from pathlib import Path
+import sys
+import os
+import geopandas as gpd
+import rasterio
+import numpy as np
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[0]}/"
+sys.path.append(path_mod)
+
+from src.indicators.drought.config import Config
+from src.utils_general.ghs import get_ghs_data
+from src.utils_general.ghs import classify_urban_areas
+
+# Setup
+
+iso3 = "mdg"
+raw_dir = os.path.join(os.environ["AA_DATA_DIR"], 'public', 'raw', iso3)
+processed_dir = os.path.join(os.environ["AA_DATA_DIR"], 'public', 'processed', iso3)
+adm4_path = os.path.join(raw_dir, 'cod_ab', 'mdg_admbnda_adm4_BNGRC_OCHA_20181031.shp')
+```
+
+Use a couple of functions to load GHS data, and then classify urban areas off of that. Can adjust GHS grid cells considered urban areas and % of raster cells per polygon required to classify polygon as urban.
+
+```python
+# GHS tile bounding boxes for MDG
+box = [(22,10),
+       (22,11),
+       (22,12)]
+
+# Download data
+
+get_ghs_data("SMOD", box, iso3, raw_dir)
+
+# Load data
+
+adm4 = gpd.read_file(adm4_path, crs='4326').to_crs('ESRI:54009')
+with rasterio.open(os.path.join(raw_dir, 'ghs', 'mdg_SMOD_2015_1km_mosaic.tif')) as src:
```

Comment:
Is there a reason you are using rasterio instead of rioxarray (or xarray)? I have generally found that rioxarray is a bit more reliable with managing coordinates and CRS's correctly (`rioxarray.open_rasterio()` could be used in this case)

Reply:
Monica mentioned it when we discussed this afternoon. The initial code I was using from the COVID work I was using was built on `rasterio`, so I based on that, but could be worth changing in the future to rioxarray!